src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -1417,18 +1419,18 @@ public GlowEntity spawn(Location location, Class<? extends GlowEntity> clazz, Sp
                 entity = constructor.newInstance(location);
                 GlowEntity impl = entity;
                 // function.accept(entity); TODO: work on type mismatches
-                EntitySpawnEvent spawnEvent;
+                EntitySpawnEvent spawnEvent = null;
                 if (entity instanceof LivingEntity) {
                     spawnEvent = EventFactory.callEvent(new CreatureSpawnEvent((LivingEntity) entity, reason));
-                } else {
+                } else if (!(entity instanceof Item)) { // ItemSpawnEvent is called elsewhere
                     spawnEvent = EventFactory.callEvent(new EntitySpawnEvent(entity));
                 }
-                if (!spawnEvent.isCancelled()) {
-                    List<Message> spawnMessage = entity.createSpawnMessage();
-                    getRawPlayers().stream().filter(player -> player.canSeeEntity(impl)).forEach(player -> player.getSession().sendAll(spawnMessage.toArray(new Message[spawnMessage.size()])));
-                } else {
+                if (spawnEvent != null && spawnEvent.isCancelled()) {
                     // TODO: separate spawning and construction for better event cancellation
                     entity.remove();
+                } else {
```

Comment:
Shouldn't this branch also require non null? Wrap the whole if else in a null check.

Reply:
No, if the event is null, it means that the event was fired elsewhere. The spawn message is still necessary.

Reply:
Oh whoops, I don't know how, but I misread that null check as `entity`, not `spawnEvent`.